src/compas_rhino/geometry/trimesh/curvature.py

```diff
@@ -0,0 +1,316 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import Rhino
+from Rhino.Geometry.Vector3d import Multiply, CrossProduct
+from Rhino.Geometry.Point3d import FromPoint3f
+
+from math import pi
+from math import sqrt
+from math import atan2
+
+from clr import StrongBox
+from scriptcontext import doc
+from System import Array
+
+from compas.plugins import plugin
+
+
+__all__ = [
+    'trimesh_gaussian_curvature',
+    'trimesh_mean_curvature',
+    'trimesh_principal_curvature',
+]
+
+
+@pluggable(category="trimesh", requires=['Rhino'])
+def trimesh_gaussian_curvature(M):
+    """Compute the discrete Gaussian curvature of a triangle mesh.
+
+    Parameters
+    ----------
+    M : (list, list)
+        A mesh represented by a list of vertices and a list of faces.
+
+    Returns
+    -------
+    list
+        The discrete Gaussian curvature per vertex.
+
+    Notes
+    -----
+    Description: The angle defect at a vertex is used to describe the Gaussian curvature in a neighborhood around a vertex.
+
+    Notation Convention:
+    :math:`K` - discrete Gaussian curvature
+    :math:`j,k` - the vertices from the Star of vertex i
+    :math:`e_{ij}, e_{ik}` - the vector from vertex i to j, i to k
+    :math:`\theta_{i}^{jk}` - interior angle at vertex i of triangle ijk
+
+    Formula:
+    .. math::
+
+        K_{i} = 2\pi-\sum\theta_{i}^{jk}
```

Comment:
Do you have references for this and the other formulas?

Reply:
Hi Dr. Lytle,

thanks for the suggestion. I will include something like this under all formulas:

```
    References
    ----------
    .. [1] Discrete Curvature by Keenan Crane, 03:16-07:11, at https://youtu.be/sokeN5VxBB8
```

Sounds good?